monai/data/image_writer.py

```diff
@@ -0,0 +1,395 @@
+            spatial_size=output_spatial_shape_,
+        )
+        data_np = data_torch[0].detach().cpu().numpy()
+        if original_channels:
+            data_np = np.moveaxis(data_np, 0, -1)  # channel last
+            data_np = data_np.reshape(list(data_np.shape[:3]) + list(original_channels))
+        else:
+            data_np = data_np[0]
+        return data_np, ensure_mat44(target_affine)
+
+    @classmethod
+    def convert_to_channel_last(
+        cls,
+        data: NdarrayOrTensor,
+        channel_dim: Optional[int] = 0,
+        squeeze_end_dims: bool = True,
+        spatial_ndim: Optional[int] = 3,
+        contiguous: bool = False,
+    ):
+        """
+        Rearrange the data array axes to make the `channel_dim`-th dim the last
+        dimension and ensure there are three spatial dimensions. If
+        ``channel_dim`` is ``None``, a new axis will be appended to the last
+        dimension.
+
+        When ``squeeze_end_dims`` is ``True``, a postprocessing step will be
+        applied to remove any trailing singleton dimensions.
+
+        Args:
+            data: input data to be converted to "channel-last" format.
+            channel_dim: specifies the axis of the data array that is the channel dimension.
+                ``None`` indicates no channel dimension.
+            squeeze_end_dims: if ``True``, any trailing singleton dimensions will be removed (after the channel
+                has been moved to the end). So if input is `(H,W,D,C)` and C==1, then it will be saved as `(H,W,D)`.
+                If D is also 1, it will be saved as `(H,W)`. If ``False``, image will always be saved as `(H,W,D,C)`.
+            spatial_ndim: modifying the spatial dims if needed, so that output to have at least
+                this number of spatial dims. If ``None``, the output will have the same number of
+                spatial dimensions as the input.
+            contiguous: if ``True``, the output will be contiguous.
+        """
+        # change data to "channel last" format
+        if channel_dim is not None:
+            # _chns = ensure_tuple_rep(channel_dim)
+            # data = moveaxis(data, _chns, tuple(range(-len(_chns), 0)))
+            data = moveaxis(data, channel_dim, -1)
+        else:  # adds a channel dimension
+            data = data[..., None]
+        # To ensure at least three spatial dims
+        if spatial_ndim:
+            while len(data.shape) < spatial_ndim + 1:  # assuming the data has spatial + channel dims
+                data = data[..., None, :]
+            while len(data.shape) > spatial_ndim + 1:
+                data = data[..., 0, :]
+        # if desired, remove trailing singleton dimensions
+        while squeeze_end_dims and data.shape[-1] == 1:
+            data = np.squeeze(data, -1)
+        if contiguous:
+            data = ascontiguousarray(data)
+        return data
+
+    @classmethod
```

Comment:
May I know why using `classmethod` here? Maybe making it into a utility function directly?

Thanks.

Reply:
the PIL writer would override this class as affine is not needed in that case